src/components/AvatarGroup/AvatarGroup.tsx

```diff
@@ -50,22 +54,28 @@ const AvatarGroup: React.FC<AvatarGroupProps> = ({
   max = 5,
   counterProps,
   counterTooltipCustomProps,
-  counterTooltipIsVirtualizedList = false
+  counterTooltipIsVirtualizedList = false,
+  padding = true
 }) => {
   const { displayAvatars, counterTooltipAvatars } = useMemo(() => {
     const childrenArray = Array.isArray(children) ? children : [children];
     return {
-      displayAvatars: childrenArray.slice(0, max),
+      displayAvatars: childrenArray.slice(0, max).map((avatar) => {
+        return React.cloneElement(avatar, {
+          ...avatar?.props,
+          className: cx(styles.avatarContainer, avatarClassName, padding && styles.noPadding)
+        });
+      }),
       counterTooltipAvatars: childrenArray.slice(max)
     };
-  }, [children, max]);
+  }, [children, max, padding]);
```

Comment:
You don't need to change padding of `Avatar`, only padding of `AvatarGroup`
Besides, if you propose to do clone here, why not to just move all the props from line 80 to here?
Let's really move it here, so it would be memoized too, but let's remove padding from here